tfhe/src/integer/gpu/mod.rs

```diff
@@ -273,6 +281,164 @@ pub unsafe fn unchecked_scalar_mul_integer_radix_kb_async<T: UnsignedInteger, B:
+        streams.gpu_indexes.as_ptr(),
+        streams.len() as u32,
+        std::ptr::addr_of_mut!(mem_ptr),
+        encryption_glwe_dimension.0 as u32,
+        encryption_polynomial_size.0 as u32,
+        compression_glwe_dimension.0 as u32,
+        compression_polynomial_size.0 as u32,
+        lwe_dimension.0 as u32,
+        ks_level.0 as u32,
+        ks_base_log.0 as u32,
+        0u32,
+        0u32,
+        0u32,
+        num_blocks,
+        message_modulus.0 as u32,
+        carry_modulus.0 as u32,
+        PBSType::Classical as u32,
+        lwe_per_glwe,
+        storage_log_modulus,
+        CompressionType::Compress as u32,
+        true,
+    );
+
+    cuda_compression_compress_integer_radix_ciphertext_64(
+        streams.ptr.as_ptr(),
+        streams.gpu_indexes.as_ptr(),
+        streams.len() as u32,
+        glwe_array_out.as_mut_c_ptr(0),
+        lwe_array_in.as_c_ptr(0),
+        fp_keyswitch_key.ptr.as_ptr(),
+        num_blocks,
+        mem_ptr,
+    );
+
+    cleanup_cuda_compression_integer_radix_ciphertext_64(
+        streams.ptr.as_ptr(),
+        streams.gpu_indexes.as_ptr(),
+        streams.len() as u32,
+        std::ptr::addr_of_mut!(mem_ptr),
+    );
+}
+
+#[allow(clippy::too_many_arguments)]
+/// # Safety
+///
+/// - [CudaStreams::synchronize] __must__ be called after this function as soon as synchronization
+///   is required
+pub unsafe fn compression_decompress_integer_radix_async<T: UnsignedInteger, B: Numeric>(
+    streams: &CudaStreams,
+    lwe_array_out: &mut CudaVec<T>,
+    glwe_in: &CudaVec<T>,
+    bootstrapping_key: &CudaVec<B>,
+    message_modulus: MessageModulus,
+    carry_modulus: CarryModulus,
+    encryption_glwe_dimension: GlweDimension,
+    encryption_polynomial_size: PolynomialSize,
+    compression_glwe_dimension: GlweDimension,
+    compression_polynomial_size: PolynomialSize,
+    lwe_dimension: LweDimension,
+    pbs_base_log: DecompositionBaseLog,
+    pbs_level: DecompositionLevelCount,
+    lwe_per_glwe: u32,
+    storage_log_modulus: u32,
+    vec_indexes: &CudaVec<u32>,
+    num_blocks: u32,
+) {
+    assert_eq!(
+        streams.gpu_indexes[0],
+        glwe_in.gpu_index(0),
+        "GPU error: all data should reside on the same GPU."
+    );
+    assert_eq!(
+        streams.gpu_indexes[0],
+        bootstrapping_key.gpu_index(0),
+        "GPU error: all data should reside on the same GPU."
+    );
+    let mut mem_ptr: *mut i8 = std::ptr::null_mut();
+    scratch_cuda_compression_integer_radix_ciphertext_64(
+        streams.ptr.as_ptr(),
+        streams.gpu_indexes.as_ptr(),
+        streams.len() as u32,
+        std::ptr::addr_of_mut!(mem_ptr),
+        encryption_glwe_dimension.0 as u32,
+        encryption_polynomial_size.0 as u32,
+        compression_glwe_dimension.0 as u32,
+        compression_polynomial_size.0 as u32,
+        lwe_dimension.0 as u32,
+        0u32,
+        0u32,
+        pbs_level.0 as u32,
+        pbs_base_log.0 as u32,
+        0,
```

Comment:
I don't think it's necessary to pass the grouping factor as argument since only classical PBS is supported for the compression.

Reply:
My intention was to keep the door open to multi-bit PBS support, but since we don't have parameters for that yet I will remove that and leave it for a next PR.